app/src/main/java/com/keylesspalace/tusky/fragment/PreferencesFragment.java

```diff
@@ -62,6 +67,31 @@ public void onCreate(Bundle savedInstanceState) {
 
         addPreferencesFromResource(preference);
 
+        Preference regexPref = findPreference("tabFilterRegex");
+        if (regexPref != null) regexPref.setOnPreferenceClickListener(preference1 -> {
+            // Reset the error dialog when shown; if the dialog was closed with the cancel button
+            // while an invalid regex was present, this would otherwise cause buggy behaviour.
+            ((EditTextPreference) regexPref).getEditText().setError(null);
+
+            // Test the regex on as the user inputs text, ensuring immediate feedback and preventing
+            // setting of an invalid regex, which would cause a crash loop.
+            ((EditTextPreference) regexPref).getEditText().addTextChangedListener(new TextWatcher() {
+                @Override
+                public void afterTextChanged(Editable s) {
+                    try {
+                        Pattern.compile(s.toString());
+                        ((EditTextPreference) regexPref).getEditText().setError(null);
+                        ((AlertDialog) ((EditTextPreference) preference1).getDialog()).getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(true);
+                    } catch (IllegalArgumentException e) {
+                        ((AlertDialog) ((EditTextPreference) preference1).getDialog()).getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false);
+                        ((EditTextPreference) regexPref).getEditText().setError("Invalid regular expression!");
+                    } catch (NullPointerException e) {}
```

Comment:
Ah, my bad. I did this as a quick workaround when noticing unexpected crashes on exiting the dialog and completely forgot to investigate further and clean up. It's also been a fair few years since I last read Effective Java! :man_shrugging: